pyam/utils.py

```diff
@@ -425,6 +425,12 @@ def format_data(df, index, **kwargs):  # noqa: C901
         short_error = short_error_regex.search(str(e)).group()
         raise_data_error(f"{short_error} in `data`", df.iloc[[row_nr]])
 
+    # check for infinite values
+    inf_rows = np.isinf(df)
```

Comment:
I don't think that `format_data` is the right place for this check. In my opinion checking if there are inf values has nothing to do with formatting data. @danielhuppmann, I understand that you suggested this place, what do you think about it?